src/Processors/Transforms/buildPushingToViewsChain.cpp

```diff
@@ -622,13 +622,23 @@ Chain buildPushingToViewsChain(
     /// Do not push to destination table if the flag is set
     else if (!no_destination)
     {
-        auto sink = storage->write(query_ptr, metadata_snapshot, context, async_insert);
-        metadata_snapshot->check(sink->getHeader().getColumnsWithTypeAndName());
-        sink->setRuntimeData(thread_status, elapsed_counter_ms);
+        try
```

Comment:
I have the same patch, but I was waiting for https://github.com/ClickHouse/ClickHouse/pull/73802 to be landed (cause I have a fix for another problem there, egh)

But my patch simply replaces `Exception` with `Poco::Exception` here - https://github.com/ClickHouse/ClickHouse/blob/208c9a2a9bcb38ee9cb0d010ecb0de7cb86a02f7/src/Processors/Transforms/buildPushingToViewsChain.cpp#L550

And your implementation will hide errors for regular INSERTs under `materialized_views_ignore_errors=1`:

```sql
insert into function url('https://0.0.0.0/foo.tsv', 'TabSeparated', 'key Int') settings http_max_tries=1, materialized_views_ignore_errors=1 values (2)
```

So let's remove this try/catch block and simply replace the catch in the reference.